hazelcast/src/main/java/com/hazelcast/nio/serialization/DefaultPortableReader.java

```diff
@@ -299,19 +331,22 @@ private int readNestedPosition(String fieldName, FieldType type) throws IOExcept
     }
 
     private int readPosition(FieldDefinition fd) throws IOException {
-        return in.readInt(offset + fd.getIndex() * 4);
+        int pos = in.readInt(offset + fd.getIndex() * Bits.INT_SIZE_IN_BYTES);
+        short len = in.readShort(pos);
+        // name + len + type
+        return pos + Bits.SHORT_SIZE_IN_BYTES + len + 1;
```

Comment:
I did not understand this part. We read the pos. I was expecting to return this pos value. Then we read a size again. It looks like we return the end position of field !?!, but this is not how we use it. What am I missing?